rust/psibase/src/native.rs

```diff
@@ -218,39 +235,36 @@ pub fn kv_get<V: PackableOwned, K: ToKey>(
 /// If one is found, and the first `match_key_size` bytes of the found key
 /// matches the provided key, then returns the value. Use [get_key_bytes] to get
 /// the found key.
-pub fn kv_greater_equal_bytes(db: crate::DbId, key: &[u8], match_key_size: u32) -> Option<Vec<u8>> {
+pub fn kv_greater_equal_bytes(db: DbId, key: &[u8], match_key_size: u32) -> Option<Vec<u8>> {
     let size =
         unsafe { native_raw::kvGreaterEqual(db, key.as_ptr(), key.len() as u32, match_key_size) };
-    if size < u32::MAX {
-        Some(get_result_bytes(size))
-    } else {
-        None
-    }
+    get_optional_result_bytes(size)
 }
 
 /// Get the key-value pair immediately-before provided key
 ///
 /// If one is found, and the first `match_key_size` bytes of the found key
 /// matches the provided key, then returns the value. Use [get_key_bytes] to get
 /// the found key.
-pub fn kv_less_than_bytes(db: crate::DbId, key: &[u8], match_key_size: u32) -> Option<Vec<u8>> {
+pub fn kv_less_than_bytes(db: DbId, key: &[u8], match_key_size: u32) -> Option<Vec<u8>> {
     let size =
         unsafe { native_raw::kvLessThan(db, key.as_ptr(), key.len() as u32, match_key_size) };
-    if size < u32::MAX {
-        Some(get_result_bytes(size))
-    } else {
-        None
-    }
+    get_optional_result_bytes(size)
 }
 
 /// Get the maximum key-value pair which has key as a prefix
 ///
 /// If one is found, then returns the value. Use [get_key_bytes] to get the found key.
-pub fn kv_max_bytes(db: crate::DbId, key: &[u8]) -> Option<Vec<u8>> {
+pub fn kv_max_bytes(db: DbId, key: &[u8]) -> Option<Vec<u8>> {
     let size = unsafe { native_raw::kvMax(db, key.as_ptr(), key.len() as u32) };
-    if size < u32::MAX {
-        Some(get_result_bytes(size))
-    } else {
-        None
-    }
+    get_optional_result_bytes(size)
+}
+
+/// Get the maximum key-value pair which has key as a prefix
+///
+/// If one is found, then returns the value. Use [get_key] to
+/// get the found key.
+pub fn kv_max<K: ToKey, V: PackableOwned>(db_id: DbId, key: &K) -> Option<V> {
+    let bytes = kv_max_bytes(db_id, &key.to_key());
+    bytes.map(|v| V::unpack(&v[..], &mut 0).unwrap()) // unwrap won't panic
```

Comment:
It will, and should, panic if the data is corrupt. e.g. someone didn't follow the rules at https://doc-sys.psibase.io/cpp-service/reference/table.html#schema-changes

Reply:
`unpacked()` is more straight-forward than `unpack()`